pages/admin/settings.tsx

```diff
@@ -0,0 +1,230 @@
+import React, { useState } from "react";
+import {
+    Flex,
+    Text,
+    Button,
+    Spacer,
+    Image,
+    useDisclosure,
+} from "@chakra-ui/react";
+import { useTable } from "react-table";
+import adminUsers from "data/adminUsers";
+import { DeleteIcon } from "@chakra-ui/icons";
+import AddModal from "./modals/addModal";
+import EditModal from "./modals/editModal";
+import DeleteModal from "./modals/deleteModal";
+
+enum ModalType {
+    ADD = "add",
+    EDIT = "edit",
+    DELETE = "delete",
+}
+
+export enum RoleType { // ensure these values are the ones in the data
+    ADMIN = "ADMIN",
+    SUPER_ADMIN = "SUPER ADMIN",
+}
+
+const SettingsPage: React.FC = () => {
+    const { isOpen, onOpen, onClose } = useDisclosure(); // controls all 3 modals
+    const [modalType, setModalType] = useState(ModalType.ADD); // determines which modal is shown when isOpen is true
+    const [selectedIndex, setSelectedIndex] = useState(0);
+    const [email, setEmail] = useState("");
+    const [role, setRole] = useState(RoleType.ADMIN);
+    const [emailError, setEmailError] = useState("");
+    const [roleError, setRoleError] = useState("");
+
+    // many states are used by multiple modals and must be set before showing a modal
+    const onAddClick = () => {
+        setModalType(ModalType.ADD);
+        setEmail("");
+        setRole(RoleType.ADMIN);
+        setEmailError("");
+        setRoleError("");
+        onOpen();
+    };
+
+    const onEditClick = (index: number) => {
+        setModalType(ModalType.EDIT);
+        setSelectedIndex(index);
+        setEmail(data[index].email);
+        setRole(data[index].role as RoleType);
+        setEmailError("");
+        setRoleError("");
+        onOpen();
+    };
+
+    const onDeleteClick = (index: number) => {
+        setModalType(ModalType.DELETE);
+        setSelectedIndex(index);
+        onOpen();
+    };
+
+    // TODO fetch real data
+    const data = React.useMemo(() => adminUsers, []);
```

Comment:
Let's move all the table stuff to a different file for the component itself? Like `<SettingsTable>` and we can also add in the colums there and get data as a prop

Reply:
100% agree